Row: Allow non-unique column names like sqlite3.Row

zmedico opened the pull request:
* keys method now returns a list of column names which are not necessarily unique

* `__getitem__` method now returns the value corresponding to the first matching column name

* items and values methods can now return multiple items or values that correspond to the same key (like multidict)

* `__str__` method now inherits the tuple implementation since dict requires unique keys

Closes: https://github.com/rqlite/pyrqlite/issues/58

otoolep commented:
Does it make sense to add a test to https://github.com/rqlite/pyrqlite/blob/master/src/test/test_dbapi.py showing that the issue is addressed?

zmedico committed:
Row: Allow non-unique column names like sqlite3.Row

* keys method now returns a list of column names which
  are not necessarily unique

* __getitem__ method now returns the value corresponding
 to the first matching column name

* items and values methods can now return multiple items
  or values that correspond to the same key (like multidict)

* __str__ method now inherits the tuple implementation
  since dict requires unique keys

Closes: https://github.com/rqlite/pyrqlite/issues/58

zmedico commented:
> Does it make sense to add a test to https://github.com/rqlite/pyrqlite/blob/master/src/test/test_dbapi.py showing that the issue is addressed?

Yes, good idea. Now I've added a join test in the style of those in test_dbapi.py. I put it in test_row.py to keep it separate from the tests imported from pysqlite.

otoolep commented:
LGTM.

This long-standing issue https://github.com/rqlite/rqlite/issues/530 seems related, FWIW.
